### PR TITLE
umpire: Backport `-fcompare-debug-second` flag removal to older versions

### PR DESCRIPTION
**Pull request by msimberg**
Backports https://github.com/LLNL/Umpire/pull/853 to 2022.10 and 2023.06 (the latter not yet added to spack, but the PR is merged after 2023.06 so it will be the upper bound).

Besides the issue described in https://github.com/LLNL/Umpire/issues/852, the flag breaks builds where umpire is built with GCC and the flag gets unconditionally addeed but a dependent library is built with nvhpc which does not support the flag at all. Since the flag has been completely removed in the development version of umpire, I'm unconditionally applying the patch to the affected versions.

### CODE REVIEW

**Review by rasolca — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
Confirmed patch exists.

**Comment by tldahlgren** [comment]
@davidbeckingsale As maintainer, do you want/need to review this PR before it is merged?

**Pull request merged**
